floris/optimization/load_optimization/load_optimization.py

```diff
@@ -0,0 +1,546 @@
+        max_dist_D (flat, optional): Maximum distance downstream of a turbine beyond which wake
+            added turbulence is assumed to be zero, in rotor diameters. Defaults to 10.0
+            (see IEC 61400-1 Ed. 4 Annex E).
+        exp_ws_std (float, optional): Exponent for the wind speed standard deviation.
+            Defaults to 1.0.
+        exp_thrust (float, optional): Exponent for the thrust. Defaults to 1.
+
+    Returns:
+        float: Value of A that satisfies the target cost/MW/findex
+
+    """
+
+    if fmodel.core.state is not State.USED:
+        raise ValueError("FlorisModel must be run before finding A for target cost/MW/findex")
+
+    # Compute farm power
+    farm_power = fmodel.get_farm_power()
+
+    # Compute farm VOC
+    farm_voc = compute_farm_voc(
+        fmodel=fmodel,
+        A=1.0,
+        load_ambient_tis=load_ambient_tis,
+        wake_slope=wake_slope,
+        max_dist_D=max_dist_D,
+        exp_ws_std=exp_ws_std,
+        exp_thrust=exp_thrust,
+    )
+
+    return 1e-6 * target_VOC_per_MW_findex / (farm_voc.sum() / (farm_power.sum()))
+
+
+def optimize_power_setpoints(
+    fmodel: FlorisModel,
+    A: float,
+    load_ambient_tis: np.array,
+    wake_slope: float = 0.3,
+    max_dist_D: float = 10.0,
+    exp_ws_std: float = 1.0,
+    exp_thrust: float = 1.0,
+    power_setpoint_initial: np.array = None,
+    derating_levels: np.array = np.linspace(1.0, 0.001, 5),
+):
+    """Optimize the derating of each turbine to maximize net revenue sequentially from upstream to
+    downstream.
+
+    Args:
+        fmodel (FlorisModel): FlorisModel object
+        A (float): Coefficient for the VOC calculation
+        load_ambient_tis (list or np.array): Ambient 'load' turbulence intensity for each findex,
+            expressed as fractions of mean wind speed
+        wake_slope (float, optional): Wake slope, defined as the lateral expansion of the wake on
+            each side per unit downstream distance along the axial direction. Defaults to 0.3.
+        max_dist_D (flat, optional): Maximum distance downstream of a turbine beyond which wake
+            added turbulence is assumed to be zero, in rotor diameters. Defaults to 10.0
+            (see IEC 61400-1 Ed. 4 Annex E).
+        exp_ws_std (float, optional): Exponent for the wind speed standard deviation.
+            Defaults to 1.0.
+        exp_thrust (float, optional): Exponent for the thrust. Defaults to 1.
+        power_setpoint_initial (np.array, optional): Initial power setpoint for each turbine.
+            If None, each turbine's rated power will be used. Defaults to None.
+        derating_levels (np.array, optional): Array of derating levels to consider in optimization,
+            represented as fractions of the power_setpoint_initial.
+            Defaults to np.linspace(1.0, 0.001, 5).
+
+    """
+
+    # Ensure we're in derating mode
+    fmodel.set_operation_model("simple-derating")
```

Comment:
Are/will there be other derating models besides "simple-derating"? If so, it would be good to document that this module only considers simple-derating.

Reply:
I'll just move to raise error and not automatically change the mode